specification/basic_concepts/basic_concepts.tex

```diff
@@ -1,79 +1,209 @@
+    \item[Service invocation] --- The communication method for one-to-one request/response
     interactions\footnote{Like remote procedure call (RPC).}.
 \end{description}
 
-For each type of communication, a predefined set of data structures is used,
-where each data structure has a unique identifier --- the \emph{data type ID} (DTID).
-Additionally, every data structure definition has a pair of major and minor semantic version numbers,
+For each type of communication, a predefined set of data types is used,
+where each data type has a unique name.
+Additionally, every data type definition has a pair of major and minor version numbers,
 which enable data type definitions to evolve in arbitrary ways while ensuring a well-defined
 migration path if backward-incompatible changes are introduced.
-Some data structures are standard and defined by the protocol specification (of which only a small 
+Some data types are standard and defined by the protocol specification (of which only a small
 subset are required); others may be specific to a particular application or vendor.
 
-Since every message or service data type has its own unique data type ID,
-and each node in the network has its own unique node ID,
-a pair of data type ID and node ID can be used to support redundant nodes with identical
-functionality inside the same network.
+\subsubsection{Subjects and services}
+
+Message exchanges between nodes are grouped into \emph{subjects} by their semantic meaning.
+Message exchanges belonging to the same subject use same message data type down to the major version
+(minor versions are allowed to differ) and pertain to same function or process within the system.
+
+Request/response exchanges between nodes are grouped into \emph{services} by their semantic meaning,
+like messages are grouped into subjects.
+Requests and their corresponding responses that belong to the same service use same service data type down to
+the major version (minor versions are allowed to differ; as a consequence, the minor data type version number
+of a service response may differ from that of its corresponding request) and pertain to same function.
+
+Each message subject is identified by a unique natural number -- a \emph{subject ID};
+likewise, each service is identified by a unique \emph{service ID}.
+An umbrella term \emph{port ID} is used to refer either to a subject ID or to a service ID
+(port identifiers have no direct manifestation in the construction of the protocol,
+but they are convenient for discussion).
+
+\subsection{Data types}
+
+\subsubsection{Data type definitions}
+
+Message and service data types\footnote{Data types are the core concept of UAVCAN.}
+are defined using the \emph{data structure description language} (DSDL) (chapter \ref{sec:dsdl}).
+A DSDL definition specifies the name, major version, minor version, the data structure,
+and an optional predefined port ID of the data type.
```

Comment:
When I first encountered "data type ids are optional" it really threw me for a loop. I'd love to get footnote or link here helping the reader understand that port IDs are eventually required but can be resolved later by a system when not using predefined identifiers. It's redundant and non-normative to mention it at this point but a footnote would help the read understand that the port ID is not, ultimately, optional.

Reply:
They are not required for nested types. I am going to move this question into a separate ticket.